### PR TITLE
 Build and unit test in Docker container

### PR DESCRIPTION
**Pull request by liptakmatyas**
-   Add a `Dockerfile` that installs all the necessary dependencies on
    top of a _Debian Buster_ image and copies the contents of the
    repository into the image.
-   Add `build-container.sh` to conveniently build the Docker image.
-   Add `compile-and-test-in-docker.sh` to build the library and run its
    unit tests inside the Docker image.
-   Add instructions to `README.md` on how to use the Docker image.

By making a contribution to this project, I certify that:

        (a) The contribution was created in whole or in part by me and I
            have the right to submit it under the GPL-3.0 license; or

        (b) The contribution is based upon previous work that, to the best
            of my knowledge, is covered under an appropriate open source
            license and I have the right under that license to submit that
            work with modifications, whether created in whole or in part
            by me, under GPL-3.0 license; or

        (c) The contribution was provided directly to me by some other
            person who certified (a), (b) or (c) and I have not modified
            it.

        (d) I understand and agree that this project and the contribution
            are public and that a record of the contribution (including all
            personal information I submit with it, including my sign-off) is
            maintained indefinitely and may be redistributed consistent with
            this project or the license(s) involved.

### CODE REVIEW

**Comment by omartijn** [comment]
I am somewhat unsure about the point of this PR. Since this repository contains a library, it doesn't serve much purpose to have a docker container with only that in it. The library is only useful when you wish to use it.

Being able to run the test from this container also doesn't really help either. You'd want to know whether the tests succeed on the same system/os that you intend to use the library. The container won't answer that for you.

I'd like to know a bit more about the idea behind this PR.

**Comment by liptakmatyas** [comment]
@omartijn The point is, that this will be the Docker image that we'll use on Travis CI.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+&&  apt-get -y install \
+        build-essential \
+        clang-8 \
+        clang-tools-8 \
+        cmake \
+        g++-8 \
+        libboost1.67-all-dev \
+        libcrypto++-dev \
+        libsodium-dev \
+&&  apt-get -y autoremove \
```
Comment (omartijn) [inline Dockerfile]: Why is the `autoremove` and `autoclean` necessary?

File: build-container.sh [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+if [[ -z ${PGPPL_BUILDER_IMAGE_NAME} ]]; then
+    PGPPL_BUILDER_IMAGE_NAME='summitto/pgp-packet-library-builder'
```
Comment (omartijn) [inline build-container.sh]: I think using getopt is a nicer way then depending on environment variables.

**Comment by omartijn** [comment]
Is this PR going anywhere?

**Comment by liptakmatyas** [comment]
My latest information is that this will be picked up again when we figure out which CI system to use for this repo. I'm not sure, if this is part of @JamesWrigley's investigation or if it will be handled separately...

**Comment by JamesWrigley** [comment]
I believe we already decided on using Travis CI because it lets us easily build Docker containers that we can run the tests in, but I might give Github Actions a quick look as well. It's not related to our internal CI discussion. I did some work on it a couple of weeks ago, but then the demo took priority so I'll pick it up when I get back.

**Comment by bysnack** [comment]
@JamesWrigley @liptakmatyas can this PR be closed now that we are using github actions?

**Comment by liptakmatyas** [comment]
I think yes, but let's have the others chime in too.

**Comment by JamesWrigley** [comment]
I guess so.

**Comment by bysnack** [comment]
Then I'm closing this PR since we are using github actions and it doesn't make sense to have it there for no use, it can be reopened in the future if we need it for whatever reason.

Pinging @vicsn and @rockhouse for visibility

**Pull request closed without merging**